src/server/src/main/java/io/cassandrareaper/management/http/HttpManagementConnectionFactory.java

```diff
@@ -113,45 +134,106 @@ private void registerConnectionsGauge() {
           .containsKey(MetricRegistry.name(HttpManagementConnectionFactory.class, "openHttpManagementConnections"))) {
         this.metricRegistry.register(
             MetricRegistry.name(HttpManagementConnectionFactory.class, "openHttpManagementConnections"),
-            (Gauge<Integer>) () -> HTTP_CONNECTIONS.size());
+            (Gauge<Integer>) HTTP_CONNECTIONS::size);
       }
     } catch (IllegalArgumentException e) {
       LOG.warn("Cannot create openHttoManagementConnections metric gauge", e);
     }
   }
 
-  private ICassandraManagementProxy connectImpl(Node node)
+  private HttpCassandraManagementProxy connectImpl(Node node)
       throws ReaperException, InterruptedException {
-    Integer managementPort = 8080; // TODO - get this from the config.
-    String rootPath = ""; // TODO - get this from the config.
     Response pidResponse = getPid(node);
     if (pidResponse.getStatus() != 200) {
       throw new ReaperException("Could not get PID for node " + node.getHostname());
     }
+    Integer managementPort = config.getHttpManagement().getManagementApiPort();
+    String rootPath = ""; // TODO - get this from the config.
 
-    String host = node.getHostname();
-
-    HTTP_CONNECTIONS.computeIfAbsent(host, new Function<String, HttpCassandraManagementProxy>() {
+    return HTTP_CONNECTIONS.computeIfAbsent(node.getHostname(), new Function<String, HttpCassandraManagementProxy>() {
       @Nullable
       @Override
       public HttpCassandraManagementProxy apply(@Nullable String hostName) {
-        DefaultApi apiClient = new DefaultApi(
-            new ApiClient().setBasePath("http://" + hostName + ":" + managementPort + rootPath));
+        ReaperApplicationConfiguration.HttpManagement httpConfig = config.getHttpManagement();
+        boolean useMtls = httpConfig.getKeystore() != null && !httpConfig.getKeystore().isEmpty();
```

Comment:
Issue: do we maybe want a check here to confirm that the truststore is also available (not just the keystore).

Reply:
I guess that is indeed the case. It won't be the common case for our users, but theoretically it could work so no need to insist on a custom trust store.